app/controllers/pension_types_controller.rb

```diff
@@ -0,0 +1,8 @@
+class PensionTypesController < ApplicationController
+  def next
+    question = GuideDecorator.cached_for(GuideRepository.new.find(params[:question_id]))
+    answer = question.metadata.answers[params[:response]]
+
+    redirect_to answer || params[:question_id]
```

Comment:
GuideRepository.new.find(params[:question_id]) will raise an error if a guide with that name is not found, so that ensures whitelisting of our guides.